File: src/responses/stats/homeStatsResponse.ts
import * as z from "zod";
import { RockItSongWithAlbumResponse } from "@/responses/rockItSongWithAlbumResponse";

export const HomeStatsResponse = z.object({
    songsByTimePlayed: z.array(RockItSongWithAlbumResponse),
    randomSongsLastMonth: z.array(RockItSongWithAlbumResponse),
    nostalgicMix: z.array(RockItSongWithAlbumResponse),
    hiddenGems: z.array(RockItSongWithAlbumResponse),
    communityTop: z.array(RockItSongWithAlbumResponse),
    monthlyTop: z.array(RockItSongWithAlbumResponse),
    moodSongs: z.array(RockItSongWithAlbumResponse),
});

File: backend/responses/stats/homeStatsResponse.py
from typing import List
from pydantic import BaseModel

from backend.responses.general.songWithAlbum import RockItSongWithAlbumResponse



class HomeStatsResponse(BaseModel):
    songsByTimePlayed: List[RockItSongWithAlbumResponse]
    randomSongsLastMonth: List[RockItSongWithAlbumResponse]
    nostalgicMix: List[RockItSongWithAlbumResponse]
    hiddenGems: List[RockItSongWithAlbumResponse]
    communityTop: List[RockItSongWithAlbumResponse]
    monthlyTop: List[RockItSongWithAlbumResponse]
    moodSongs: List[RockItSongWithAlbumResponse]
